### DISCUSSION TITLE
Upgrade Dagger code to use new 'functions' style workflow

### DISCUSSION THREAD

**Discussion by MDr164** [9elements/firmware-action#239] [General]
Dagger recently switched around the way pipelines are registered for use with their updated CLI. This is arguably not needed for Dagger to work correctly but it would enable us to use the new goodies the CLI offers, like built-in pipeline tracing and maybe even Dagger Cloud for people that don't want to rely on GH actions as the backend. But this would require a slight refactor on the layout of the pipeline steps as they now require a 'dagger' folder to be in and they'll have to be methods on a global struct. I think otherwise the majority stays the same. I didn't open this as an issue as I wanted to collect pros/cons first of actually doing this.

**Comment by AtomicFS** [comment]
Personally, after our in-person discussion, I think this is a good idea. Let's do this.

I opened a issue #269.
